libs/offline-tasks/src/main/java/org/opensearch/offline_tasks/task/Task.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.offline_tasks.task;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * A Background Task to be run on Offline Node.
+ */
+@ExperimentalApi
+public class Task {
+
+    /**
+     * Task identifier used to uniquely identify a Task
+     */
+    private final TaskId taskId;
+
+    /**
+     * Various params to used for Task execution
+     */
+    private final TaskParams params;
+
+    /**
+     * Type/Category of the Task
+     */
+    private final TaskType taskType;
+
+    /**
+     * Constructor for Task
+     *
+     * @param taskId Task identifier
+     * @param params Task Params
+     * @param taskType Task Type
+     */
+    public Task(TaskId taskId, TaskParams params, TaskType taskType) {
+        this.taskId = taskId;
```

Comment:
Why are we not autogenerating the taskId? How will collisions be avoided in case the taskId is generated by the caller?

Reply:
i need to think about this. As per initial thoughts, the caller should generate a TaskId and make sure it is not interfering/colliding with other tasks. 

But to absolutely avoid collisions our implementation would have check on duplicate Task submission and throw an error if duplicate task is created